src/main/java/org/carlspring/cloud/storage/s3fs/S3AccessControlList.java

```diff
@@ -18,61 +13,55 @@
 
     private final String key;
 
-    private final Iterable<Grant> grants;
-
-    private final Owner owner;
 
+    /**
+     * Creates a new S3AccessControlList
+     *
+     * @param fileStoreName
+     * @param key
+     * @param grants unused
+     * @param owner unused
+     * @deprecated use {@link #S3AccessControlList(String, String)}
+     */
+    @Deprecated
     public S3AccessControlList(final String fileStoreName,
                                final String key,
-                               final Iterable<Grant> grants,
-                               final Owner owner)
+                               final Iterable<Grant> grants, //unused, but keeping to preserve signature
+                               final Owner owner //unused, but keeping to preserve signature
+    )
     {
         this.fileStoreName = fileStoreName;
-        this.grants = grants;
         this.key = key;
-        this.owner = owner;
-    }
-
-    public String getKey()
-    {
-        return key;
     }
 
     /**
-     * have almost one of the permission set in the parameter permissions
+     * Creates a new S3AccessControlList
      *
-     * @param permissions almost one
-     * @return
+     * @param fileStoreName
+     * @param key
      */
-    private boolean hasPermission(final EnumSet<Permission> permissions)
+    public S3AccessControlList(final String fileStoreName, final String key)
+    {
+        this.fileStoreName = fileStoreName;
+        this.key = key;
+    }
+
+    public String getKey()
     {
-        return StreamSupport.stream(grants.spliterator(), false)
-                            .anyMatch(grant -> StringUtils.equals(grant.grantee().id(), owner.id()) &&
-                                               permissions.contains(grant.permission()));
+        return key;
     }
 
     public void checkAccess(final AccessMode[] modes)
             throws AccessDeniedException
     {
         for (AccessMode accessMode : modes)
         {
+            // Checking the ACL grants is not sufficient to determine access as bucket policy may override ACL.
+            // Any permission problems will have to be handled at time of access.
             switch (accessMode)
             {
                 case EXECUTE:
                     throw new AccessDeniedException(fileName(), null, "file is not executable");
```

Comment:
`switch` statement has one case label only, and should probably be replaced with an `if` statement.

```suggestion
            if (accessMode == AccessMode.EXECUTE)
            {
                throw new AccessDeniedException(fileName(), null, "file is not executable");
            }
```